subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -795,9 +789,8 @@ uint8_t ull_adv_sync_pdu_alloc(struct ll_adv_set *adv,
 
 #if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
 	/* Get reference to new periodic advertising PDU data buffer */
-	if ((hdr_add_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) ||
-	    (!(hdr_rem_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) &&
-	     ed_prev)) {
+	if (extra_data_flag == ULL_ADV_PDU_EXTRA_DATA_ALLOC_ALWAYS ||
+	    (!(extra_data_flag == ULL_ADV_PDU_EXTRA_DATA_ALLOC_NEVER) && ed_prev)) {
```

Comment:
Semantically I think this makes more sense:
```suggestion
	if (extra_data_flag == ULL_ADV_PDU_EXTRA_DATA_ALLOC_ALWAYS ||
	    (extra_data_flag == ULL_ADV_PDU_EXTRA_DATA_ALLOC_IF_EXIST && ed_prev)) {
```

Reply:
Changed according to suggestion. Thanks!